src/Features/Core/Portable/IntroduceVariable/AbstractIntroduceParameterService.cs

```diff
@@ -0,0 +1,656 @@
+                                                              string parameterName, SyntaxEditor editor,
+                                                              bool allOccurrences, CancellationToken cancellationToken)
+        {
+            var generator = SyntaxGenerator.GetGenerator(semanticDocument.Document);
+            var matches = FindMatches(semanticDocument, expression, semanticDocument, scope, allOccurrences, cancellationToken);
+            var parameterNameSyntax = (TIdentifierNameSyntax)generator.IdentifierName(parameterName);
+            // Parenthesize the variable, and go and replace anything we find with it.
+            // NOTE: we do not want elastic trivia as we want to just replace the existing code 
+            // as is, while preserving the trivia there.  We do not want to update it.
+
+            var replacement = generator.AddParentheses(parameterNameSyntax, includeElasticTrivia: false)
+                                         .WithAdditionalAnnotations(Formatter.Annotation);
+
+            var syntaxFacts = semanticDocument.Document.GetRequiredLanguageService<ISyntaxFactsService>();
+
+            foreach (var match in matches)
+            {
+                if (match.Equals(expression))
+                {
+                    if (IsExpressionRemovable(semanticDocument, expression))
+                    {
+                        var localDeclaration = expression.FirstAncestorOrSelf<SyntaxNode>(node => syntaxFacts.IsLocalDeclarationStatement(node));
+                        if (localDeclaration is not null)
+                        {
+                            editor.RemoveNode(localDeclaration);
+                        }
+                    }
+                    else
+                    {
+                        var expressionToken = syntaxFacts.GetIdentifierOfIdentifierName(parameterNameSyntax);
+                        var updatedExpressionToken = expressionToken.WithAdditionalAnnotations(RenameAnnotation.Create());
+                        parameterNameSyntax = parameterNameSyntax.ReplaceToken(expressionToken, updatedExpressionToken);
+
+                        replacement = generator.AddParentheses(parameterNameSyntax, includeElasticTrivia: false)
+                                                     .WithAdditionalAnnotations(Formatter.Annotation);
+                        editor.ReplaceNode(match, replacement);
+                    }
+                }
+                else
+                {
+                    editor.ReplaceNode(match, replacement);
+                }
+            }
+        }
+
+        private int GetInsertionIndex(SemanticDocument document, Compilation compilation, ISyntaxFactsService syntaxFacts, SyntaxNode methodDeclaration, CancellationToken cancellationToken)
+        {
+            var parameterList = syntaxFacts.GetParameterList(methodDeclaration);
+            var parameterSymbolList = GetParameterList(document, parameterList!, cancellationToken);
```

Comment:
insetad of `!` you can do `Contract.ThrowIfNull` .   i don't trust that we're not actually getting null back in these cases.

Reply:
Changhed